compiler/rustc_attr_parsing/src/attributes/cfg.rs

```diff
@@ -77,14 +74,33 @@ fn gate_cfg(gated_cfg: &GatedCfg, cfg_span: Span, sess: &Session, features: &Fea
     }
 }
 
+pub trait CfgEval {
+    fn eval_individual(&mut self, condition: Condition, features: Option<&Features>) -> bool;
+
+    fn eval_accessible_path(
+        &mut self,
+        target_crate: rustc_span::Ident,
+        target_path_segs: impl Iterator<Item = rustc_span::Ident>,
+    ) -> Option<bool>;
+}
+
 /// Evaluate a cfg-like condition (with `any` and `all`), using `eval` to
 /// evaluate individual items.
 pub fn eval_condition(
     cfg: &ast::MetaItemInner,
     sess: &Session,
     features: Option<&Features>,
-    eval: &mut impl FnMut(Condition) -> bool,
+    eval: &mut impl CfgEval,
 ) -> bool {
+    eval_condition_inner(cfg, sess, features, eval).unwrap_or_default()
+}
+
+fn eval_condition_inner(
```

Comment:
I'm not sure why this is necessary if `None` always turns into `false` anyway.

Reply:
My intention was to make `#[cfg(not(invalid_cfg))]` and `#[cfg(invalid_cfg)]` both acts as `#[cfg(FALSE)]`.  I can leave comment in the code if it's not obvious enough.